jadx-core/src/test/java/jadx/core/xmlgen/ResNameUtilsTest.java

```diff
@@ -0,0 +1,89 @@
+	void testSanitizeAsResourceName(String name, String postfix, boolean allowNonPrintable, String expectedResult) {
+		assertThat(ResNameUtils.sanitizeAsResourceName(name, postfix, allowNonPrintable)).isEqualTo(expectedResult);
+	}
+
+	@DisplayName("Check convertToJavaIdentifier(name)")
+	@ParameterizedTest(name = "{0} -> {1}")
+	@MethodSource("provideArgsForConvertToJavaIdentifierTest")
+	void testConvertToJavaIdentifier(String name, String expectedResult) {
+		assertThat(ResNameUtils.convertToJavaIdentifier(name)).isEqualTo(expectedResult);
+	}
+
+	private static Stream<Arguments> provideArgsForSanitizeAsResourceNameTest() {
+		return Stream.of(
+				Arguments.of("name", "_postfix", false, "name"),
+
+				Arguments.of("/name", "_postfix", true, "_name_postfix"),
+				Arguments.of("na/me", "_postfix", true, "na_me_postfix"),
+				Arguments.of("name/", "_postfix", true, "name__postfix"),
+
+				Arguments.of("$name", "_postfix", true, "_name_postfix"),
+				Arguments.of("na$me", "_postfix", true, "na_me_postfix"),
+				Arguments.of("name$", "_postfix", true, "name__postfix"),
+
+				Arguments.of(".name", "_postfix", true, "_.name_postfix"),
+				Arguments.of("na.me", "_postfix", true, "na.me"),
+				Arguments.of("name.", "_postfix", true, "name."),
+
+				Arguments.of("0name", "_postfix", true, "_0name_postfix"),
+				Arguments.of("na0me", "_postfix", true, "na0me"),
+				Arguments.of("name0", "_postfix", true, "name0"),
+
+				Arguments.of("-name", "_postfix", true, "_name_postfix"),
+				Arguments.of("na-me", "_postfix", true, "na_me_postfix"),
+				Arguments.of("name-", "_postfix", true, "name__postfix"),
+
+				Arguments.of("Ĉname", "_postfix", false, "_name_postfix"),
+				Arguments.of("naĈme", "_postfix", false, "na_me_postfix"),
+				Arguments.of("nameĈ", "_postfix", false, "name__postfix"),
+
+				Arguments.of("Ĉname", "_postfix", true, "Ĉname"),
+				Arguments.of("naĈme", "_postfix", true, "naĈme"),
+				Arguments.of("nameĈ", "_postfix", true, "nameĈ"),
+
+				// Uncomment this when XID_Start and XID_Continue characters are correctly determined.
+				// Arguments.of("Жname", "_postfix", true, "Жname"),
+				// Arguments.of("naЖme", "_postfix", true, "naЖme"),
+				// Arguments.of("nameЖ", "_postfix", true, "nameЖ"),
+				//
+				// Arguments.of("€name", "_postfix", true, "_name_postfix"),
+				// Arguments.of("na€me", "_postfix", true, "na_me_postfix"),
+				// Arguments.of("name€", "_postfix", true, "name__postfix"),
+
+				Arguments.of("", "_postfix", true, "_postfix"),
+
+				Arguments.of("if", "_postfix", true, "if_postfix"),
+				Arguments.of("default", "_postfix", true, "default_postfix"),
+				Arguments.of("true", "_postfix", true, "true_postfix"),
+				Arguments.of("_", "_postfix", true, "__postfix"));
+	}
+
+	private static Stream<Arguments> provideArgsForConvertToJavaIdentifierTest() {
+		return Stream.of(
+				Arguments.of("ThemeDesign", "ThemeDesign"),
+				Arguments.of("Theme.Design", "Theme_Design"),
```

Comment:
dot on style name it is normal, you don't need replace it to _

Reply:
This test suite refers to the `convertToJavaIdentifier` function, which returns a java identifier.

Reply:
Okay, what about names inside style.xml?

Reply:
In resource names, a dot can be used anywhere except as the first character.